packages/core/src/application.ts

```diff
@@ -193,6 +206,7 @@ export class Application extends Context {
 
 export interface ApplicationConfig {
   components?: Array<Constructor<Component>>;
+  servers?: Array<Function | {}>;
```

Comment:
Can we use real types instead of `Function` and `{}`? I am thinking about `Array<Constructor<Server> | NamedServer>`.

I think the name+ctor format is too verbose, I am proposing to change `servers` into a key/value object instead.

```ts
// before
{
 servers: [RestServer, {name: 'Server2', ctor: RestServer}]
}

// now
{
 servers: {RestServer, Server2: RestServer}
}
```

Thoughts?

Reply:
At `getServer` (https://github.com/strongloop/loopback-next/pull/576/files/66ca975deb6d9a54d45c5b80a11fd77b827a3fdd..64b101de3a36b8236bcd724da9f35c6720afaaa2#diff-b108b10ab51c865c712388b56d86f520R129), I had to alter the code to use typeof instead of instanceof since it wasn't always correctly identifying the type of the input (if you change that section back, the new tests I wrote fail).

A similar approach was needed here since I can't check for instanceof `Constructor` here, which means I need to fall back on `typeof item === 'function'` and thus use the more general type.

>I think the name+ctor format is too verbose, I am proposing to change servers into a key/value object instead.
```js
// now
{
 servers: {RestServer, Server2: RestServer}
}
```
Did you mean something like this?
```js
{
  servers: {
    Server1: RestServer,
    Server2: RestServer,
  },
}
```

Reply:
> At getServer, I had to alter the code to use typeof instead of instanceof since it wasn't always correctly identifying the type of the input (if you change that section back, the new tests I wrote fail).

What was the `instanceof` check? `Server` is an interface, it cannot be used at runtime (for `instanceof` checks for example). I think `instanceof` check will be never reliable as long as the there may be multiple copies of a single module in `node_modules` tree, because you can have multiple instances of `RestServer` constructor function (one instance per each module copy).

OTOH, what I was proposing here, was to tell typescript to require that the `servers` array contains constructors of classes inheriting from `Server`, that should be checked at runtime and should not require `instanceof` operator.

What am I missing?

> A similar approach was needed here since I can't check for instanceof Constructor here, which means I need to fall back on typeof item === 'function' and thus use the more general type.

I think we need to distinguish between runtime checks (where you want to decide whether the item is a constructor function or a named config entry, and you cannot use `instanceof`) and type definitions used by TypeScript to verify that `servers` array contains only valid entries. In my comment, I was trying to address the latter.

> Did you mean something like this?
```
    {
      servers: {
        Server1: RestServer,
        Server2: RestServer,
      },
    }
```

Yes. We can leverage ES6 feature where you can say `{RestServer}` as a short-hand for `{RestServer: RestServer}`.